### PR TITLE
Pass NULL in the host parameter of grpc_channel_create_call

### PR DESCRIPTION
**Pull request by makdharma**
Pass NULL in the host parameter of grpc_channel_create_call

### CODE REVIEW

**Comment by jcanizales** [comment]
Please ammend the PR description to point to the correct issue :)

**Comment by makdharma** [comment]
Sorry, typo fixed. Fixes #2642.

**Comment by makdharma** [comment]
test this please

**Comment by jcanizales** [comment]
Of course, with Jenkins running 1.0 now, tests here fail... The ideal solution is to do in `build_tests.sh` the same as in `travis.yaml`: Uninstall Cocoapods and install the specific version we use. That way the tool goes with the source. For the moment, for this PR we can just run the Travis tests.

**Comment by makdharma** [comment]
test this please

**Comment by makdharma** [comment]
test this please

**Comment by jcanizales** [comment]
It worked :)

File: src/objective-c/GRPCClient/private/GRPCChannel.m [hunk 1]
```diff
@@ -199,9 +199,7 @@ - (grpc_call *)unmanagedCallWithPath:(NSString *)path
                                   NULL, GRPC_PROPAGATE_DEFAULTS,
                                   queue.unmanagedQueue,
                                   path.UTF8String,
-                                  // Get "host" from "host:port"
-                                  // TODO(jcanizales): Use NSURLs throughout, to clarify these.
-                                  [_host componentsSeparatedByString:@":"][0].UTF8String,
```
Comment (jcanizales) [inline src/objective-c/GRPCClient/private/GRPCChannel.m]: As `_host` was only used for this, please delete the ivar.

Comment (makdharma) [inline src/objective-c/GRPCClient/private/GRPCChannel.m]: _host is also used for Cronet channel creation.

Comment (jcanizales) [inline src/objective-c/GRPCClient/private/GRPCChannel.m]: Oh!

**Comment by jtattermusch** [comment]
@jcanizales Please don't blindly mark pull requests as "ready to merge" if a huge portion of tests is red and the required statuses (Portability tests and Interop tests are not green).
Without the "Required" statuses being green, the merge button is disabled and I physically cannot merge the PR even if I wanted to - so marking as "ready to merge" basically means that I will be sitting on your pull request, rerunning tests until they are green - and that's not something I'd want to do. I am fully aware that Jenkins can be pretty flaky (and I spent lots and lots of time trying to fix that), but right now we that's the way things are so we have to bite the bullet and accept that. Please note that I have incoming merge requests from 20 other people and if everyone was just marking everything as ready to merge without checking the tests results, I'd spend my entire day just checking test results and rerunning tests. (I already spend much more time than I'd wish doing that). Thanks.

**Comment by jcanizales** [comment]
Sorry, forgot about the merge button getting disabled. Next time I'll rerun the flakes before adding the label. Feel free to remove the label and reassign to me if I forget again!

**Comment by jtattermusch** [comment]
Thanks, I appreciate that.

**Pull request merged**